controller/api/destination/server.go

```diff
@@ -364,17 +364,44 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 	return nil
 }
 
-func (s *server) createAddress(pod *corev1.Pod, port uint32) (watcher.Address, error) {
+// getPortForPod returns the port that a `pod` is listening on.
+//
+// If the `host` address given is the HostIP, use the HostPort mapping for the `pod`
+// to determine the port to return.
+func (s *server) getPortForPod(pod *corev1.Pod, host string, port uint32) (uint32, error) {
+	if host == pod.Status.PodIP {
+		return port, nil
+	}
+
+	if host == pod.Status.HostIP {
+		for _, container := range pod.Spec.Containers {
+			for _, containerPort := range container.Ports {
+				if uint32(containerPort.HostPort) == port {
+					return uint32(containerPort.ContainerPort), nil
+				}
+			}
+		}
+	}
+
```

Comment:
Do we have to guard against `.Status` or `.Spec` being null? This can crash at runtime with bad pod data.

Reply:
The linter says they can't be nil. Do we trust that?

Reply:
Ok after checking the source type it does seem like `Status`/`Spec` can't be nil. (Though I guess `pod` could be nil).

Reply:
I've added a *bob ross voice* happy little nil check